### PR TITLE
cargo: bump trust-dns-client and trust-dns-proto to 0.22

### PR DESCRIPTION
**Pull request by siretart**

### CODE REVIEW

**Review by flouthoc — COMMENTED** [review]
Something seems off with PR all the tests fail for me on local with this

**Comment by flouthoc** [comment]
@siretart Do you mind if i take over this PR ? I am experimenting with this and i see some issues.

**Comment by siretart** [comment]
No not at all, please do drive this home!

**Comment by flouthoc** [comment]
Okay I think this should pass now and should also fix the issues we are seeing on intermittent drops on some setups, I think it was issue way we were using sender stream and newer version of `trust-dns` have forced to creat dedicated stream for each peer.

**Comment by Luap99** [comment]
> Okay I think this should pass now and should also fix the issues we are seeing on intermittent drops on some setups, I think it was issue way we were using sender stream and newer version of `trust-dns` have forced to creat dedicated stream for each peer.

Lets hope for the best. Code LGTM if it passes the tests.

Thanks @siretart and @flouthoc

**Comment by flouthoc** [comment]
okay tests are green but hold let me becnhmark this as well.
/hold

**Comment by flouthoc** [comment]
/hold cancel 

okay this is good to go . @mheon @baude PTAL

File: src/dns/coredns.rs [hunk 1]
```diff
@@ -124,7 +127,8 @@ impl CoreDns {
                         Ok(msg) => {
                             let src_address = msg.addr();
                             let mut dns_resolver = self.resolv_conf.clone();
-                            let sender = sender.clone();
+                            let sender = sender_original.clone();
+                            let sender = sender.with_remote_addr(src_address);
```
Comment (mheon) [inline src/dns/coredns.rs:131]: Any reason this can't be combined into line 130?

Comment (flouthoc) [reply src/dns/coredns.rs]: done.

**Comment by flouthoc** [comment]
@containers/netavark-maintainers PTAL

**Comment by siretart** [comment]
I took the liberty of uploading this code change to Debian as [1.4.0-2](https://packages.debian.org/source/testing/aardvark-dns) because of https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1024980 -- and I have seen reports that this patch makes DNS resolution actually work for podman.

**Pull request merged**